Amazon.IonObjectMapper.Test/CustomIonSerializerTest.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using static Amazon.IonObjectMapper.Test.Utils;
+
+namespace Amazon.IonObjectMapper.Test
+{
+    [TestClass]
+    public class CustomIonSerializerTest
+    {
+        [TestMethod]
+        public void CanUseACustomSerializerFactory()
+        {
+            var customSerializer = new IonSerializer(new IonSerializationOptions { CustomContext = new Dictionary<string, object>() 
+                {
+                    { "customCourseSerializer", new CustomCourseSerializerFactory()}
+                }
+            });
+            var stream = customSerializer.Serialize(TestObjects.bob);
+            var serialized = StreamToIonValue(stream);
+            Assert.IsTrue(serialized.ContainsField("name"));
+            Assert.IsTrue(serialized.ContainsField("course"));
+            Assert.IsTrue(serialized.ContainsField("id"));
+            var course = serialized.GetField("course");
+            Assert.AreEqual(9, course.GetField("Sections").IntValue);
```

Comment:
We should also validate the time.

Reply:
Added.

Reply:
I think you should use a time that won't change, using `Now` introduces potential flakiness due to the fact that it may change between when you deserialize it and when you test it.

Reply:
Make sense. Changed to a fixed time.